macros/path_transformations/transform_paths.sql

```diff
@@ -0,0 +1,126 @@
+        {% endif %}
+        {{target.schema}}.{{path_transform_name}}(
+      {% endfor %}
+
+      transformed_path
+
+      {% for _, transform_param in var('path_transforms').items()|reverse %}
+        {% if transform_param %}, '{{transform_param}}' {% endif %}
+        )
+      {% endfor %}
+
+      as transformed_path
+
+    {% else %}
+     transformed_path
+    {% endif %}
+
+  from {{ source_cte }}
+
+  )
+
+{% endmacro %}
+
+
+{% macro databricks__transform_paths(model_type, source_cte) %}
+
+  {% set total_transformations = var('path_transforms').items()|length %}
+  -- set loop_count using namespace to define it as global variable for the loop to work
+  {% set loop_count = namespace(value=1) %}
+
+  {% for path_transform_name, transform_param in var('path_transforms').items() %}
+
+    {%- if loop_count.value == 1 %}
+      {% set previous_cte = 'arrays' %}
+    {% else %}
+      {% set previous_cte = loop_count.value-1 %}
+    {% endif %}
+
+    , transformation_{{ loop_count.value|string }} as (
+
+      select
+        customer_id,
+        {% if model_type == 'conversions' %}
+        conversion_tstamp,
+        revenue,
+        {% endif %}
+        path,
+        {% if path_transform_name == 'unique_path' %}
+          {{ path_transformation('unique_path') }} as transformed_path
+
+        {% elif path_transform_name == 'frequency_path' %}
+          {{ path_transformation('frequency_path', '') }} as transformed_path
+
+        {% elif path_transform_name == 'first_path' %}
+          {{ path_transformation('first_path') }} as transformed_path
+
+        {% elif path_transform_name == 'exposure_path' %}
+          {{ path_transformation('exposure_path', '') }} as transformed_path
+
+        {% elif path_transform_name == 'remove_if_not_all' %}
+          {{ path_transformation('remove_if_not_all', transform_param) }} as transformed_path
+
+        {% elif path_transform_name == 'remove_if_last_and_not_all' %}
+          {{ path_transformation('remove_if_last_and_not_all', transform_param) }} as transformed_path
+
+        {% else %}
+          {%- do exceptions.raise_compiler_error("Snowplow Error: the path transform - '"+path_transform_name+"' - is not supported. Please refer to the Snowplow docs on tagging. Please use one of the following: exposure_path, first_path, frequency_path, remove_if_last_and_not_all, remove_if_not_all, unique_path") %}
+        {% endif %}
+
+        {%- if loop_count.value == 1 %}
+         from {{ source_cte }}
+         )
+        {% else %}
+         from transformation_{{ previous_cte|string }}
+         )
+        {% endif %}
+        {% set loop_count.value = loop_count.value + 1 %}
+        {% set previous_cte = loop_count.value-1 %}
+
```

Comment:
Would it be clearer if we swapped lines 102 and 103? So:
`{% set previous_cte = loop_count.value %}
{% set loop_count.value = loop_count.value + 1 %}`
Otherwise we increment just to deincrement (_apparently deincrement isn't a word, but i'm sticking with it_) in the next line

Reply:
Yeah, definitely. I think decrement is the word here.  :)